### PR TITLE
winpr/file: extend API

### PR DESCRIPTION
**Pull request by MartinHaimberger**
Add function GetFileHandleForFileDescriptor to get
a file handle for a valid file descriptor.

### CODE REVIEW

File: winpr/libwinpr/file/file.c [hunk 1]
```diff
@@ -215,13 +216,34 @@ HANDLE GetStdHandle(DWORD nStdHandle)
...
 }
 
 #endif /* _WIN32 */
 
+/* Extended API */
+
+HANDLE GetFileHandleForFileDescriptor(int fd)
+{
+#ifdef WIN32
+    return INVALID_HANDLE_VALUE;
```
Comment (mfleisz) [inline winpr/libwinpr/file/file.c]: I'd call _get_osfhandle (https://msdn.microsoft.com/en-us/library/ks2530z6.aspx) here

Comment (bmiklautz) [inline winpr/libwinpr/file/file.c]: @mfleisz is that still available on every windows version? - I wasn't sure.

Comment (mfleisz) [inline winpr/libwinpr/file/file.c]: @bmiklautz Since Windows 95: https://msdn.microsoft.com/en-us/library/ks2530z6(v=vs.80).aspx

**Comment by mfleisz** [comment]
LGTM +1

**Comment by bmiklautz** [comment]
+1 for the nice changes on windows. @mfleisz good idea.
@MartinHaimberger thx.

**Pull request merged**